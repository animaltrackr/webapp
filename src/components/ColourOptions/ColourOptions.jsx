import React, { Component } from 'react';
import Typography from 'antd/lib/typography';
import Select from 'antd/lib/select';
import Spin from 'antd/lib/spin';
import Icon from 'antd/lib/icon';

import './ColourOptions.less';

const { Text } = Typography;
const { Option } = Select;

class ColourOptions extends Component {
	onColourChange = (deer, value) => {
		this.props.onStateChange(deer, value, 'colour');
	};

	render() {
		const loadingIcon = <Icon type="loading" style={{ fontSize: 20 }} spin />;

		return (
			<div className="option">
				<Text className="option-title">Colour:</Text>
				<Select
					defaultValue={this.props.deer ? this.props.deer.colour : ''}
					onChange={this.onColourChange.bind(this, this.props.deer)}
				>
					<Option value={'blue'}>blue</Option>
					<Option value={'gray'}>gray</Option>
					<Option value={'green'}>green</Option>
					<Option value={'purple'}>purple</Option>
					<Option value={'red'}>red</Option>
					<Option value={'white'}>white</Option>
					<Option value={'yellow'}>yellow</Option>
				</Select>
				<div
					className="spin-icon"
					hidden={this.props.deer ? !this.props.deer.loading.colour : true}
				>
					<Spin indicator={loadingIcon} />
				</div>
			</div>
		);
	}
}

ColourOptions.propTypes = {};

export default ColourOptions;
