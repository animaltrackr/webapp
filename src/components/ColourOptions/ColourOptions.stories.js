import React from 'react';
import { storiesOf } from '@storybook/react';
import ColourOptions from './ColourOptions';

storiesOf('ColourOptions', module).add('ColourOptions', () => (
	<div>
		<ColourOptions />
	</div>
));
