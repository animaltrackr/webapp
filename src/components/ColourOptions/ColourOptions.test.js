import React from 'react';
import { shallow } from 'enzyme';
import ColourOptions from './ColourOptions';

describe('<ColourOptions />', () => {
	test('renders', () => {
		const wrapper = shallow(<ColourOptions />);
		expect(wrapper).toMatchSnapshot();
	});
});
