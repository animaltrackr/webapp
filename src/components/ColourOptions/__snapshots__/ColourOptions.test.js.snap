// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColourOptions /> renders 1`] = `ShallowWrapper {}`;
